compiler/rustc_lint/src/nonstandard_style.rs

```diff
@@ -437,12 +437,14 @@ impl<'tcx> LateLintPass<'tcx> for NonSnakeCase {
             if let hir::Node::Pat(parent_pat) = cx.tcx.hir().get(cx.tcx.hir().get_parent_node(hid))
             {
                 if let PatKind::Struct(_, field_pats, _) = &parent_pat.kind {
-                    for field in field_pats.iter() {
-                        if field.ident != ident {
-                            // Only check if a new name has been introduced, to avoid warning
-                            // on both the struct definition and this pattern.
-                            self.check_snake_case(cx, "variable", &ident);
-                        }
+                    if field_pats
+                        .iter()
+                        .find(|field| !field.is_shorthand && field.pat.hir_id == p.hir_id)
+                        .is_some()
```

Comment:
This could be `.any(|field| !field.is_shorthand && field.pat.hir_id == p.hir_id)`